src/engine/rendering/Model.cpp

```diff
@@ -151,6 +151,12 @@ namespace lei3d
 		// only select the first texture
 		aiString str;
 		mat->GetTexture(type, 0, &str);
+		std::pair<const aiTexture*, int> texdata = scene->GetEmbeddedTextureAndIndex(str.C_Str());
```

Comment:
How did you find this function?? Very nice

https://github.com/assimp/assimp/blob/master/include/assimp/scene.h#L408

Reply:
Couldn't find it when I was searching around for a potential fix to this bug.

Reply:
Knowing the gltf format specification helps: https://github.com/KhronosGroup/glTF/blob/main/specification/2.0/figures/gltfOverview-2.0.0d.png.  
Specifically with this, we're working with gltf 2.0.  
Also some digging through assimp github in the issues tab, because 99% chance someone's had that/similar problem before